dartsim/src/Base.hh

```diff
@@ -488,17 +493,37 @@ class Base : public Implements3d<FeatureList<Feature>>
     }
   }
 
-  public: inline std::size_t AddJoint(DartJoint *_joint)
+  public: inline std::size_t AddJoint(DartJoint *_joint,
+      const std::string &_fullName, std::size_t _modelID)
   {
     const std::size_t id = this->GetNextEntity();
-    this->joints.idToObject[id] = std::make_shared<JointInfo>();
+    auto jointInfo = std::make_shared<JointInfo>();
+    this->joints.idToObject[id] = jointInfo;
+    jointInfo->joint = _joint;
+    // The name of the dart Joint during creation is assumed to be the
+    // Gazebo-specified name.
+    jointInfo->name = _joint->getName();
+
     this->joints.idToObject[id]->joint = _joint;
```

Comment:
nit: I think this can be removed now since `jointInfo->joint` is set above.

Reply:
Updated in af9f7cca.